recipes/djinni-support-lib/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class DjinniSuppotLib(ConanFile):
+    name = "djinni-support-lib"
+    homepage = "https://djinni.xlcpp.dev"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Djinni is a tool for generating cross-language type declarations and interface bindings"
+    topics = ("java", "Objective-C", "Android", "iOS")
+    license = "Apache-2.0"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+                "fPIC": [True, False],
+                "target": ["JNI", "OBJC", "Auto"],
+                "system_java": [True, False]
+               }
+    default_options = {"shared": False,
+                        "fPIC": True ,
+                        "target": "Auto",
+                        "system_java": False
+                       }
+    exports_sources = ["patches/**", "CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def objc_support(self):
+        if self.options.target == "Auto":
+            return self.settings.os in ["iOS", "Macos"]
+        else:
+            return self.options.target == "OBJC"
+
+    @property
+    def jni_support(self):
+        if self.options.target == "Auto":
+            return self.settings.os != "iOS"
+        else:
+            return self.options.target == "JNI"
+
+
+    def configure(self):
+        if self.settings.compiler == 'Visual Studio':
+            del self.options.fPIC
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def build_requirements(self):
+        if not self.options.system_java:
+            self.build_requires("zulu-openjdk/11.0.8@")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(f"djinni-support-lib-{self.version}", self._source_subfolder)
```

Comment:
CCI still needs to support older version of python

```suggestion
        os.rename(self.name + "-" + self.version", self._source_subfolder)
```